<img src="https://userweb.fct.unl.pt/~jmc.xavier/MSI/iLogos/nsst.jpg" width="400" height="100">

# DEMI | NOVA SST

## Mecânica dos Sólidos I

# III. Materiais elásticos
## Lei de Hooke. Parte I
## Problema 2 <a name="PIII61"></a>

A roseta de extensómetros que se mostra na Figura permite obter os seguintes valores para as extensões na superfície de uma peça,

\begin{equation*}
\begin{aligned}
\varepsilon_1 &= 600 \mu\varepsilon \\
\varepsilon_2 &= 450 \mu\varepsilon \\
\varepsilon_3 &= -75 \mu\varepsilon \\
\end{aligned}
\end{equation*}

[a)](#A) Determine a extensão na direção do eixo $x$ indicado.

[b)](#B) Obtenha o tensor das tensões no ponto onde estão colados os extensómetros, considerando que o material da peça tem $E = 70$ GPa e $G = 26$ GPa.

<img src="https://userweb.fct.unl.pt/~jmc.xavier/MSI/Notebooks/Au06/Au06_P002_i1.jpg"
width="300" height="240">



## Resolução

#### a) Determine a extensão na direção do eixo $x$ indicado.<a name="A"></a> ([enunciatum](#PIII61))

A extensão linear numa determinada direção definida pelo ângulo $\alpha$ com a direção horizontal do eixo dos $xx$ é dada por:

\begin{equation*}
\begin{aligned}
\varepsilon_{\alpha} &= \varepsilon_{xx}\cos^2\alpha +
\varepsilon_{yy}\sin^2\alpha + \gamma_{xy}\sin\alpha\cos\alpha \\
\end{aligned}
\end{equation*}

Numa primeira fase, esta equação de transformação pode ser desenvolvida para a obtenção de um sistema de equações cuja solução permita a determinação das componentes cartesianas do tensor das deformações no plano de análise:

- $\theta_1 = 30^\circ$ ($\varepsilon_1 = 600 \mu\varepsilon$)
- $\theta_2 = -30^\circ$ ($\varepsilon_2 = 450 \mu\varepsilon$)
- $\theta_3 = 90^\circ$ ($\varepsilon_3 = -75 \mu\varepsilon$)

Conhecida a matriz do tensor de deformações no sistema de coordenadas $S(0,x,y)$ é possível determinar a extensão linear em qualquer direção definida pelo seu versor.

In [14]:
import numpy as np
from sympy.solvers import solve
from sympy import symbols

exx,eyy,gxy,t = symbols('exx eyy gxy t')

# unit: degree, strains = 10^(-6) micro-strains
eps1 = 600.*1e-6; ang1 = 30. # extensómetro 1
eps2 = 450.*1e-6; ang2 = -30. # extensómetro 2
eps3 = -75.*1e-6; ang3 = 90. # extensómetro 3

def extens(exx,eyy,gxy,t): return exx*np.cos(t)**2 + eyy*np.sin(t)**2 + gxy*np.sin(t)*np.cos(t)

eq1 = eps1 - extens(exx,eyy,gxy,np.deg2rad(ang1))
eq2 = eps2 - extens(exx,eyy,gxy,np.deg2rad(ang2))
eq3 = eps3 - extens(exx,eyy,gxy,np.deg2rad(ang3))
defs = solve({eq1,eq2,eq3},[exx,eyy,gxy],)
exxP  = float(defs[exx])
eyyP  = float(defs[eyy])
gxyP  = float(defs[gxy])
print('versão 1: ------------------------------')
print(f'exxP (sistema equações) = {exxP*1e6:.2f} micro def')
print(f'eyyP (sistema equações) = {eyyP*1e6:.2f} micro def')
print(f'gxyP (sistema equações) = {gxyP*1e6:.2f} micro def')
print(f'          exyP (gxyP\\2) = {gxyP*1e6/2:.2f} micro def')

versão 1: ------------------------------
exxP (sistema equações) = 725.00 micro def
eyyP (sistema equações) = -75.00 micro def
gxyP (sistema equações) = 173.21 micro def
          exyP (gxyP\2) = 86.60 micro def


Matriz das deformações no ponto $P$:

\begin{equation*}
\underline{\underline{D}} \equiv [D] \equiv D_{ij} =
\begin{bmatrix}
\varepsilon_{xx} & \varepsilon_{xy} & \varepsilon_{xz} \\
\varepsilon_{yx} & \varepsilon_{yy} & \varepsilon_{yz} \\
\varepsilon_{zx} & \varepsilon_{zy} & \varepsilon_{zz} \\
\end{bmatrix} =
\begin{bmatrix}
\varepsilon_{xx} & \gamma_{xy}/2 & \gamma_{xz}/2 \\
\gamma_{yx}/2 & \varepsilon_{yy} & \gamma_{yz}/2 \\
\gamma_{zx}/2 & \gamma_{zy}/2 & \varepsilon_{zz} \\
\end{bmatrix} =
\begin{bmatrix}
725.0 & 86.6 & 0 \\
86.6 & -75.0 & 0 \\
0 & 0 & \varepsilon_{zz}
\end{bmatrix} [\mu\varepsilon]
\end{equation*}

O valor da extensão na direção do eixo dos $xx$ resulta diretamente: $\varepsilon(0^\circ) \equiv \varepsilon_{xx} = 725.0 \mu\varepsilon$

In [15]:
angX = 0
epsX = extens(exxP,eyyP,gxyP,np.deg2rad(angX))
print(f'epsX(theta=0) = {epsX*1e6} micro-def')

epsX(theta=0) = 725.0 micro-def


#### b) Obtenha o tensor das tensões no ponto onde estão colados os extensómetros, considerando que o material da peça tem $E = 70$ GPa e $G = 26$ GPa.<a name="B"></a>
([enunciatum](#PIII61))

**Da lei de Hooke para um material isotrópico**

Para um material isotrópico a lei de Hooke tridimensional vem expressa por:

\begin{equation*}
\varepsilon_{ij} = \frac{1}{E}[(1+\nu)\sigma_{ij} - \nu\delta_{ij}\sigma_{kk}]
\end{equation*}

ou reciprocamente,

\begin{equation*}
\sigma_{ij} = \frac{E}{1+\nu}\varepsilon_{ij} + \frac{\nu E}{(1+\nu)(1-2\nu)}\varepsilon_{kk}\delta_{ij}
\end{equation*}


Em forma matricial a lei constitutiva linear elástica expressa em termos de módulo de Young e coeficiente de Poisson
vem:

\begin{equation*}
\begin{aligned}
\varepsilon_{11} &= \displaystyle\frac{1}{E}[\sigma_{11} - \nu(\sigma_{22}+\sigma_{33})] \\
\varepsilon_{22} &= \displaystyle\frac{1}{E}[\sigma_{22} - \nu(\sigma_{11}+\sigma_{33})] \\
\varepsilon_{33} &= \displaystyle\frac{1}{E}[\sigma_{33} - \nu(\sigma_{11}+\sigma_{22})] \\
2\varepsilon_{23} &= \gamma_{23} = \displaystyle\frac{\tau_{23}}{G}\\
2\varepsilon_{13} &= \gamma_{13} = \displaystyle\frac{\tau_{13}}{G}\\
2\varepsilon_{12} &= \gamma_{12} = \displaystyle\frac{\tau_{12}}{G}\\
\end{aligned}
\end{equation*}

com,

\begin{equation*}
G = \frac{E}{2(1+\nu)}
\end{equation*}

Na forma matricial têm-se:

\begin{equation*}
\begin{bmatrix}
\varepsilon_{11} \\ \varepsilon_{22} \\ \varepsilon_{33} \\ 2\varepsilon_{23} \\ 2\varepsilon_{13} \\ 2\varepsilon_{12}
\end{bmatrix} = \frac{1}{E}
\begin{bmatrix}
1 & -\nu & -\nu & 0 & 0 & 0 \\
-\nu & 1 & -\nu & 0 & 0 & 0 \\
-\nu & -\nu & 1 & 0 & 0 & 0 \\
0 & 0 & 0 & 2(1+\nu) & 0 & 0 \\
0 & 0 & 0 & 0 & 2(1+\nu) & 0 \\
0 & 0 & 0 & 0 & 0 & 2(1+\nu) \\
\end{bmatrix}
\begin{bmatrix}
\sigma_{11} \\ \sigma_{22} \\ \sigma_{33} \\ \tau_{23} \\  \tau_{13} \\ \tau_{12}
\end{bmatrix}
\end{equation*}

ou reciprocamente,

\begin{equation*}
\begin{bmatrix}
\sigma_{11} \\ \sigma_{22} \\ \sigma_{33} \\ \tau_{23} \\  \tau_{13} \\ \tau_{12}
\end{bmatrix} = \frac{E}{(1+\nu)(1-2\nu)}
\begin{bmatrix}
1-\nu & \nu & \nu & 0 & 0 & 0 \\
\nu & 1-\nu & \nu & 0 & 0 & 0 \\
\nu & \nu & 1-\nu & 0 & 0 & 0 \\
0 & 0 & 0 & (1-2\nu)/2 & 0 & 0 \\
0 & 0 & 0 & 0 & (1-2\nu)/2 & 0 \\
0 & 0 & 0 & 0 & 0 & (1-2\nu)/2 \\
\end{bmatrix}
\begin{bmatrix}
\varepsilon_{11} \\ \varepsilon_{22} \\ \varepsilon_{33} \\ 2\varepsilon_{23} \\ 2\varepsilon_{13} \\ 2\varepsilon_{12}
\end{bmatrix}
\end{equation*}

A segunda lei de termodinâmica requer que as constantes elásticas verifiquem os seguintes limites:

\begin{equation*}
\begin{aligned}
E,~G > 0 \\
-1 < \nu < \frac{1}{2}
\end{aligned}
\end{equation*}

Para a condição de estado plano de tensão têm-se:

\begin{equation*}
\begin{bmatrix}
\sigma_{11} \\ \sigma_{22} \\ \tau_{12}
\end{bmatrix} = \frac{E}{1-\nu^2}
\begin{bmatrix}
1 & \nu & 0 \\
\nu & 1 & 0 \\
0 & 0 & 1-\nu \\
\end{bmatrix}
\begin{bmatrix}
\varepsilon_{11} \\ \varepsilon_{22} \\ \gamma_{12}
\end{bmatrix}
\end{equation*}

In [16]:
E = 70*1e3 # unit: MPa
G = 26*1e3
v = E/2/G - 1
print('Propriedades do material: ---------------')
print('E (MPa) = ',E)
print(f'      v = {v:.2f}')
print(f'G (MPa) = {G:.2f}')


Propriedades do material: ---------------
E (MPa) =  70000.0
      v = 0.35
G (MPa) = 26000.00


In [17]:

# Versão 1
print('=====================================================')
print('Versão 1:')
print('=====================================================')
e11,e22,e33,e23,e13,e12 = symbols('e11 e22 e33 e23 e13 e12')
s11,s22,s12 = symbols('s11 s22 s12')

# estado plano de tensão
szz = 0 ; sxz = 0; syz = 0

def e11(sxx,syy,szz,E): return (sxx - v*(syy+szz))/E
def e22(sxx,syy,szz,E): return (syy - v*(sxx+szz))/E
def e33(sxx,syy,szz,E): return (szz - v*(sxx+syy))/E
def e23(syz,G): return (syz/G)/2
def e13(sxz,G): return (sxz/G)/2
def e12(sxy,G): return (sxy/G)/2

ex1 = exxP - e11(s11,s22,szz,E)
ex2 = eyyP - e22(s11,s22,szz,E)
ex6 = gxyP/2 - e12(s12,G)

tens1 = solve({ex1,ex2},{s11,s22})
sxxP = float(tens1[s11])
print(f'Sxx(P) [MPa] = {sxxP:.6f}')
syyP = float(tens1[s22])
print('Syy(P) [MPa] = {syyP:.6f}')

tens2 = solve({ex6},{s12})
sxyP = float(tens2[s12])
print('Sxy(P) [MPa] = {sxyP:.6f}')

ezzP = -(v/E)*(sxxP + syyP)
print('Ezz(P) = {ezzP:.4e}')

Versão 1:
Sxx(P) [MPa] = 55.594118
Syy(P) [MPa] = {syyP:.6f}
Sxy(P) [MPa] = {sxyP:.6f}
Ezz(P) = {ezzP:.4e}


In [18]:

# Versão 2
print('=====================================================')
print('Versão 2:')
print('=====================================================')
SxxPv2 = E/(1-v**2)*(exxP + v*eyyP)
print('Sxx(P) [MPa] = ','{:.6f}'.format(SxxPv2))
SyyPv2 = E/(1-v**2)*(v*exxP + eyyP)
print('Syy(P) [MPa] = ','{:.6f}'.format(SyyPv2))
SxyPv2 = G*gxyP
print('Sxy(P) [MPa] = ','{:.6f}'.format(SxyPv2))

Versão 2:
Sxx(P) [MPa] =  55.594118
Syy(P) [MPa] =  13.994118
Sxy(P) [MPa] =  4.503332


Matriz das tensões (estado plano de tensão):

\begin{equation*}
\underline{\underline{\sigma}} = [\sigma] = \sigma_{ij} =
\begin{bmatrix}
55.594 & 4.503 & 0\\
4.503 & 13.994 & 0\\
0 & 0 & 0
\end{bmatrix}~\textrm{[MPa]}
\end{equation*}

---

Copyright (c) [Mecânica dos Sólidos I - DEMI - NOVA-SST]

Interactive computing by <a href="https://jupyter.org/" target="_blank"> <span
style="color:#333399"> Jupyter Notebook </span> </a> &nbsp;|&nbsp;Coded by <a href = "mailto: jmc.xavier@fct.unl.pt">José Xavier</a>

Licensed under  <a href="http://creativecommons.org/licenses/by-sa/4.0/"
target="_blank"> <span style="color:#333399;font-size: 20px"> CC BY-SA 4.0  </span></a>
